config/config.go

```
@@ -66,21 +55,90 @@ var defaultConfig = Config{
 }
 
 func LoadConfig(file_path string) (*Config, error) {
+	config := &Config{}
+
+	if file_path == "" {
+		file_path = "./config.yaml"
+	}
+
 	content, e := os.ReadFile(file_path)
 	if e != nil {
-		return nil, e
+		return config, e
```

Is there a special reason we're returning an empty struct rather than nil? 

I don't think I mind, but I think I'd rather have it NPE if the error is ignored, rather than having an empty config

---

I tried to keep with your conventions but as you noted this was a bit too large of a PR so I expect there might be some of these. 

I'll revert this line.

---

Fixed